scripts/iql-lesson-sync/src/iql_lesson_sync/upload.py

```diff
@@ -0,0 +1,180 @@
+import shutil
+import tempfile
+import os
+from pathlib import Path
+from getpass import getpass
+import random
+import requests
+import sys
+from yaspin import yaspin
+from yaspin.spinners import Spinners
+
+
+class Lesson:
+    def __init__(self, lesson_path, lesson_id):
+        self.path = Path(lesson_path)
+        self.name = self.path.parts[-1]
+        self.id = lesson_id
+        self.zip_path = None
+
+    def zip(self):
+        self.zip_path = Path(
+            shutil.make_archive(
+                Path(tempfile.gettempdir(), "iql_lesson_sync", self.name),
+                "zip",
+                root_dir=self.path,
+            )
+        )
+
+    def delete_zip(self):
+        if self.zip_path and self.zip_path.exists():
+            os.remove(self.zip_path)
+
+
+class API:
+    def __init__(self, name, api_url, website_url, hide_urls=False):
+        self.name = name
+        self.url = api_url
+        self.website_url = website_url
+        self.auth_header = {"Authorization": f"Bearer {self.get_access_token()}"}
+        self.hide_urls = hide_urls
+
+    def get_access_token(self):
+        """
+        Either token is in env variable, or we use login details to get temporary
+        access token.
+        """
+        # Highlight where we're pushing to avoid accidental push to production
+        color = "\033[93m" if self.name == "production" else "\033[94m"
+
+        if os.environ.get("LEARNING_API_TOKEN", None) is not None:
+            print(f'✅ Found token for {color}\033[1m{self.name}\033[0m')
+            return os.environ.get("LEARNING_API_TOKEN")
+
+        print(f'🔑 Log into {color}\033[1m{self.name}\033[0m:')
+        response = requests.post(
+            f"{self.url}/auth/login",
+            json=(
+                {
+                    "email": input("   Email: "),
+                    "password": getpass("   Password: "),
+                }
+            ),
+        )
+        if response.status_code != 200:
+            print("❌ Couldn't log in 😕")
+            sys.exit()
+
+        print(f'✅ Logged in')
```

Comment:
Nit: consider running Black on this project. While we won't enforce it in CI since this isn't a Python repo, most Qiskit projects are using Black.

I use `pipx install black`, then it adds Black to your PATH.